t/format-datetime.t

```diff
@@ -0,0 +1,22 @@
+use strict;
+use warnings;
+
+use Test::More 0.96;
+use Test::Fatal;
+
+use DateTime::Format::Strptime;
+use DateTime;
+
+my $strptime = DateTime::Format::Strptime->new(
+    pattern  => '%B %Y',
+    locale   => 'pt',
+    on_error => 'croak',
+);
+
+like(
+    exception { $strptime->format_datetime('somestring') },
+    qr/Validation failed for type named DateTime declared in package DateTime::Format::Strptime::Types/,
+    'format_datetime() receives invalid type param'
```

Comment:
Should be something like `format_datetime() checks that it received a DateTime object` - test names should describe the expected behavior of the thing you're testing.